### ISSUE TITLE
Cannot compile automatically

### ISSUE THREAD

**Issue by dipterix** [igraph/rigraph#553]
The package cannot compile on ~~OSX~~ (Windows, corrected) since the `glpk` header file is missing (happened on Github Action)

The fix is easy with brew. However, when checking `systemRequirements` in the `DESCRIPTION`, it says `glpk` is optional. 

```
glpk (>= 4.57, optional)
```

I remember `igraph` used to have an embedded version of `glpk`, hence it's ok to be optional. But it's gone now because CRAN might have asked for it. Since the embedded version has been removed, should `glpk` be considered hard-requirement?

Normally `systemRequirements` is less respected, but the R-lib Github Action and R-universe use this information to install system prerequisites. Unlike CRAN, these services might only install the bare minimum sys requirements. Setting a library as optional may cause compiling errors.

**Comment by szhorvat** [comment]
At least on Linux it's not uncommon that the shared library is installed, but the header is not, as the header is found in the "development package". Is it possible that something like that is happening here?

**Comment by ntamas** [comment]
This is indeed a possibility and we can solve it by looking for `glpk.h` explicitly in the `configure` script, although it would be great if @dipterix could confirm that this is indeed the case.

**Comment by dipterix** [comment]
@szhorvat If the package can run without glpk, then that optional sys requirement makes sense. 

I'm afraid your guess is right. Since the package can proceed with compilation, then `-lglpk` flag must be detected. However, the presence of that flag is not sufficient for `<glpk.h>` to be found.

**Comment by ntamas** [comment]
Also, it seems like we _do_ check for `glpk.h` in the `configure` script so that's not an issue for Linux or macOS. My log did not show it because we do not bother to check for it if the library itself is not found.

@szhorvat This one does look fishy to me, though:

```
if test "x$enable_gmp" != "xno"; then
    AC_CHECK_LIB([glpk], [glp_read_mps], [
...
```

Note that we are testing `$enable_gmp` first, but then we look for `glpk`. I guess that's a typo, right?

**Comment by szhorvat** [comment]
I agree, it looks like a typo.

**Comment by ntamas** [comment]
Typo fixed in 827097a3
